downloader/src/main/kotlin/vcs/GitWorkingTree.kt

```diff
@@ -70,18 +70,16 @@ open class GitWorkingTree(
             SubmoduleWalk.forIndex(repo).use { walk ->
                 while (walk.next()) {
                     val path = "$prefix${walk.path}"
-                    paths += path
 
                     if (walk.repository == null) {
                         logger.info {
                             "Git submodule at '$path' not initialized. Cannot recursively list its submodules."
                         }
-
-                        continue
-                    }
-
-                    walk.repository.use { submoduleRepo ->
-                        listSubmodules(path, submoduleRepo, paths)
+                    } else {
+                        paths += path
```

Comment:
The code looks ok to me now, but the commit message doesn't:

This is not about clearing a VCS path, but about only adding paths to *initialized* submodules. So I'd say:

```
GitWorkingTree: Only add paths to initialized submodules

Do not add paths to uninitialized submodules. As these are not present in
the working tree, they should not count towards the provenance information
of that is going to be scanned.
```
```

Reply:
Unrelated, but could you please add an additional commit that changes the above `info`-log to `warn`? With the rationale that it should be an exceptional case that submodules are not initialized, as it might hide stuff from getting scanned.